### PR TITLE
Issue 14672 bulk actions popups

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/util/ActionletUtil.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+     *
+     * @param action
+     * @return
+     */
+    public static boolean hasPushPublishActionlet(final WorkflowAction action) {
+        try {
+            final List<WorkflowActionClass> actionlets = APILocator.getWorkflowAPI()
+                    .findActionClasses(action);
+            for (WorkflowActionClass actionlet : actionlets) {
+                if (actionlet.getActionlet() != null && actionlet.getActionlet().getClass()
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/util/ActionletUtil.java]: this must be change to
```
for (final WorkflowActionClass actionClass : actionClasses) {

            final Actionlet actionlet = AnnotationUtils.
                    getBeanAnnotation(ReflectionUtils.getClassFor(actionClass.getClazz()), Actionlet.class);

                if (null != actionlet && actionlet.pushPublish()) {
                        return true;
        }
```

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java [hunk 2]
```diff
@@ -2094,17 +2102,21 @@ private BulkActionsResultView fireBulkActionsTaskForSelectedInodes(final Workflo
...
+			final AdditionalParamsBean additionalParamsBean)
 			throws DotDataException {
 
 		try {
-			final String preparedBulkActionsQuery = LuceneQueryUtils.prepareBulkActionsQuery(luceneQuery);
-			Logger.debug(getClass(),"luceneQuery: "+ preparedBulkActionsQuery);
-			final List<Contentlet> contentlets = findContentletsToProcess(preparedBulkActionsQuery,
+			final String sanitizedQuery = LuceneQueryUtils
+					.sanitizeBulkActionsQuery(luceneQuery);
+			Logger.debug(getClass(), "luceneQuery: " + sanitizedQuery);
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java]: remember Logger.debug(getClass(), ()->"luceneQuery: " + sanitizedQuery);

**Review by jdotcms — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/util/ActionletUtil.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import com.dotmarketing.portlets.workflows.model.WorkflowAction;
+import com.dotmarketing.portlets.workflows.model.WorkflowActionClass;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+import java.util.List;
+
+public class ActionletUtil {
+
+    /**
+     *
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/util/ActionletUtil.java]: doc me

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/util/ActionletUtil.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        } catch (DotDataException e) {
+            Logger.error(ActionletUtil.class, String.format(
+                    "Can't determine if action '%s' has PushPublishActionlet.",
+                    action.getName()), e);
+        }
+        return false;
+    }
+
+    /**
+     *
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/util/ActionletUtil.java]: doc me

**Review by jdotcms — APPROVED** [review]

**Pull request merged**